### PR TITLE
Adds language support

### LINKED ISSUE #8: Enable languages when importing ContentTypes

### LINKED ISSUE #11: Rewrite the target adapter
EzPlatformAdapter::send() needs a little bit of rewrite, because the current code is neither pretty nor easily extendable.

### PR DESCRIPTION
**Pull request by htollefsen**
Aims to resolve #8.
Will also resolve #11 for now.

### CODE REVIEW

**Comment by ghost** [comment]
@htollefsen This branch has conflicting changes since the two PR merges I recently did.

File: src/Transfer/EzPlatform/Adapter/EzPlatformAdapter.php [hunk 1]
```diff
@@ -52,6 +53,8 @@ class EzPlatformAdapter implements TargetAdapterInterface, LoggerAwareInterface
      */
     protected $contentTypeManager;
 
+    protected $languageManager;
```
Comment (ghost) [inline src/Transfer/EzPlatform/Adapter/EzPlatformAdapter.php]: I think this should have been moved to `Transfer\EzPlatform\Repository\ObjectService` (see line 72). The same applies for `$contentTypeManager`.

File: src/Transfer/EzPlatform/Data/LanguageObject.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                throw new \Exception(sprintf('Default languagename for code "%s" not found.', $this->code));
+            }
+        }
+        $this->name = $name;
+    }
+
+    /**
+     * @var array
+     */
+    protected $defaultNameMapping = array(
```
Comment (ghost) [inline src/Transfer/EzPlatform/Data/LanguageObject.php:25]: Is there a way we can get this array from some eZ Publish classes?

File: src/Transfer/EzPlatform/Adapter/EzPlatformAdapter.php (hunk 1 above)
Comment (htollefsen) [inline src/Transfer/EzPlatform/Adapter/EzPlatformAdapter.php]: Absolutely, I'll do that.

File: src/Transfer/EzPlatform/Data/LanguageObject.php (hunk 2 above)
Comment (htollefsen) [inline src/Transfer/EzPlatform/Data/LanguageObject.php:25]: I couldn't find any, but I'll see if I can get the names from somewhere else.

Comment (htollefsen) [inline src/Transfer/EzPlatform/Data/LanguageObject.php:25]: eZSystems confirmed this on Slack (https://ezcommunity.slack.com/archives/ezplatform-use/p1451227067001211). I don't see any other way for now than to have them predefined.

File: src/Transfer/EzPlatform/Repository/Manager/LanguageManager.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use Transfer\EzPlatform\Data\LanguageObject;
+
+/**
+ * Content type manager.
+ *
+ * @author Harald Tollefsen <harald@netmaking.no>
+ */
+class LanguageManager implements LoggerAwareInterface
```
Comment (ghost) [inline src/Transfer/EzPlatform/Repository/Manager/LanguageManager.php]: Is there any reason that `LanguageManager` is not implementing the `Transfer\Data\ObjectInterface\CreatorInterface`?

Comment (htollefsen) [inline src/Transfer/EzPlatform/Repository/Manager/LanguageManager.php]: I guess theres no real good reason, it just that `LanguageObject` doesn't implement `ObjectInterface` either. Want me to rewrite these parts? I don't see any big cons regarding the current implementation, so it's for you to decide.

Comment (ghost) [inline src/Transfer/EzPlatform/Repository/Manager/LanguageManager.php]: I think we should have those classes implement the right interfaces - for the sake of uniformity in relation to other similar classes.

**Comment by htollefsen** [comment]
@valisj The interfaces has been implemented. Anything else you would like this PR to cover before we merge it?

**Comment by ghost** [comment]
@htollefsen Looks great, thanks! :+1:

**Pull request merged**